agent/agent.go

```diff
@@ -487,11 +487,12 @@ func (a *Agent) runTask(ctx context.Context, tc *taskContext) (bool, error) {
 	defer a.killProcs(ctx, tc, false)
 	defer tskCancel()
 
-	heartbeat := make(chan string, 1)
-	go a.startHeartbeat(tskCtx, tskCancel, tc, heartbeat)
-
 	innerCtx, innerCancel := context.WithCancel(tskCtx)
 
+	// Pass in idle timeout context(innerCtx) to heartbeat to enforce the idle timeout.
```

Comment:
I don't think we need to specify `innerCtx` here since it's understood to be the idle timeout context. If it's not, then we should probably rename the `innerCtx` to a better name like `idleTimeoutCtx`.